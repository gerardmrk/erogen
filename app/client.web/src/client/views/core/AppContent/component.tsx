import * as React from "react";
import styles from "./styles.scss";
import { LocalProps, StoreProps, DispatchProps } from ".";
import { Switch } from "react-router";
import LoadingDisplay from "./LoadingDisplay";
import routeConfs from "@client/views/conf.routes";
import EnhancedRoute from "@client/views/components/EnhancedRoute";

type Props = LocalProps & StoreProps & DispatchProps;

type State = {};

export class AppContent extends React.PureComponent<Props, State> {
  public render() {
    return (
      <main className={styles.main}>
        <React.Suspense fallback={<LoadingDisplay />}>
          <Switch>
            {routeConfs.map((r, i) => (
              <EnhancedRoute key={i} {...r} />
            ))}
          </Switch>
        </React.Suspense>
      </main>
    );
  }
}

export default AppContent;
