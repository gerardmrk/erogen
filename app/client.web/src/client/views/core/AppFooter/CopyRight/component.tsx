import * as React from "react";
import styles from "./component.styles.scss";
import { LocalProps, StoreProps, DispatchProps } from ".";

type Props = LocalProps & StoreProps & DispatchProps;

type State = {};

export class CopyRight extends React.PureComponent<Props, State> {
  public render() {
    return (
      <div className={styles.main}>
        <div className={styles.container}>{"CopyRight"}</div>
      </div>
    );
  }
}
