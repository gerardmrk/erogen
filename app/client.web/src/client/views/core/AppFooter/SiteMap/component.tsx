import * as React from "react";
import styles from "./styles.scss";
import { LocalProps, StoreProps, DispatchProps } from ".";

type Props = LocalProps & StoreProps & DispatchProps;

type State = {};

export class SiteMap extends React.PureComponent<Props, State> {
  public render() {
    return (
      <div className={styles.main}>
        <div className={styles.container}>{"SiteMap"}</div>
      </div>
    );
  }
}

export default SiteMap;
