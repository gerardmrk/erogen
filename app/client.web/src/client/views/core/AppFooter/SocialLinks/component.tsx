import * as React from "react";
import styles from "./component.styles.scss";
import { LocalProps, StoreProps, DispatchProps } from ".";

type Props = LocalProps & StoreProps & DispatchProps;

type State = {};

export class SocialLinks extends React.PureComponent<Props, State> {
  public render() {
    return (
      <div className={styles.main}>
        <div className={styles.container}>{"SocialLinks"}</div>
      </div>
    );
  }
}

export default SocialLinks;
