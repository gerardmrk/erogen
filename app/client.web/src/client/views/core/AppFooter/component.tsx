import * as React from "react";
import { Pane } from "evergreen-ui";
import styles from "./styles.scss";
import { LocalProps, StoreProps, DispatchProps } from ".";
import Container from "@client/views/components/Container";

type Props = LocalProps & StoreProps & DispatchProps;

type State = {};

export class AppFooter extends React.PureComponent<Props, State> {
  public render() {
    return (
      <Pane className={styles.main} border={"muted"} background={"tint3"}>
        <Container className={styles.container}>{"AppFooter"}</Container>
      </Pane>
    );
  }
}

export default AppFooter;
