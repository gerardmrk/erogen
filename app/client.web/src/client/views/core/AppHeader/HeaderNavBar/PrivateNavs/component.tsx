import * as React from "react";
import styles from "./component.styles.scss";
import { LocalProps, StoreProps, DispatchProps } from ".";
import UserDropdown from "./UserDropdown";
// import Menu from "@client/views/components/ui.collections/Menu";

type Props = LocalProps & StoreProps & DispatchProps;

type State = {};

export class PrivateNavs extends React.PureComponent<Props, State> {
  public render() {
    return (
      <div className={styles.main}>
        <div className={styles.navs}>{"..."}</div>
        <div className={styles.userdropdown}>
          <UserDropdown />
        </div>
      </div>
    );
  }
}
