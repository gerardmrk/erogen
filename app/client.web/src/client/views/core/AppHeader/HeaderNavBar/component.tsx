import * as React from "react";
import styles from "./styles.scss";
import { LocalProps, StoreProps, DispatchProps } from ".";
import PrivateNavs from "./PrivateNavs";
import PublicNavs from "./PublicNavs";

export type Props = LocalProps & StoreProps & DispatchProps;
export type State = {};

export class HeaderNavBar extends React.Component<Props, State> {
  public render() {
    return (
      <div className={styles.main}>
        {this.props.isAuthenticated ? <PrivateNavs /> : <PublicNavs />}
      </div>
    );
  }
}

export default HeaderNavBar;
