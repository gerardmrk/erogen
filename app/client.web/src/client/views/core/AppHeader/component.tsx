import * as React from "react";
import styles from "./styles.scss";
import { Pane } from "evergreen-ui/esm/layers";
import { LocalProps, StoreProps, DispatchProps } from ".";
import Container from "@client/views/components/Container";
import Logo from "./Logo";
import HeaderNavBar from "./HeaderNavBar";

export type Props = LocalProps & StoreProps & DispatchProps;

export type State = {};

export class AppHeader extends React.PureComponent<Props, State> {
  public render() {
    return (
      <Pane
        borderColor={"x"}
        is={"header"}
        border={"muted"}
        background={"blueTint"}
      >
        <Container className={styles.container}>
          <div className={styles.logo}>
            <Logo />
          </div>
          <HeaderNavBar />
        </Container>
      </Pane>
    );
  }
}

export default AppHeader;
