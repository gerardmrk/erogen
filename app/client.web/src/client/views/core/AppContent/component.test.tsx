// Only enable snapshotting when component has stabilised and is not
// expected to undergo much development changes in the future.
const SNAPSHOT_ENABLED = true;

import * as React from "react";
import { ReactTestRenderer, ReactTestInstance } from "react-test-renderer"; // prettier-ignore
import { createRenderer } from "@client/views/_fixtures_/create-renderer";

import { AppContent } from "./component";
import { MemoryRouter, withRouter } from "react-router";

const Component = withRouter(AppContent);

describe("<AppContent/>", () => {
  let renderer: ReactTestRenderer;
  let component: ReactTestInstance;

  beforeEach(() => {
    renderer = createRenderer(
      <MemoryRouter>
        <Component />
      </MemoryRouter>,
    );
    component = renderer.getInstance() as ReactTestInstance;
  });

  if (SNAPSHOT_ENABLED) {
    test("snapshot", () => {
      expect(renderer.toJSON()).toMatchSnapshot();
    });
  }

  it("renders ok", () => {
    expect(component).toBeDefined();
  });
});
