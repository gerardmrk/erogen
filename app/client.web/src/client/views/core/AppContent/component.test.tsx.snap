// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppContent/> snapshot 1`] = `
<main
  className="main"
>
  <div
    className="routeContainer"
  >
    <div
      className="main"
    >
      <div
        className="main"
      >
        <section
          className="main"
        >
          <div
            className="ui container container"
          >
            <div
              className="canvas"
            >
              <div
                className="leftPane"
              >
                <h1>
                  <em>
                    hero.heading
                  </em>
                </h1>
                <div
                  className="fluff"
                >
                  <br />
                  <p>
                    Alpha bravo charlie delta echo foxtrot golf
                  </p>
                  <br />
                  <p>
                    Hotel india juliet kilo lima mike november oscar papa
                  </p>
                  <br />
                  <p>
                    Quebec romeo sierra tango unicorn victor whiskey x-ray yankee zulu
                  </p>
                </div>
              </div>
              <div
                className="rightPane"
              >
                <img
                  src="astro.svg"
                />
              </div>
            </div>
          </div>
        </section>
        <section
          className="main"
        >
          <div
            className="ui container container"
          >
            <div
              className="brandCarousel"
            >
              <div
                className="brandItem"
              >
                <p>
                  Smashing
                </p>
              </div>
              <div
                className="brandItem"
              >
                <p>
                  Pumpkins
                </p>
              </div>
              <div
                className="brandItem"
              >
                <p>
                  Radiohead
                </p>
              </div>
              <div
                className="brandItem"
              >
                <p>
                  Chilli
                </p>
              </div>
              <div
                className="brandItem"
              >
                <p>
                  Peppers
                </p>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </div>
</main>
`;
